routers/routes/routes.go

```diff
@@ -503,6 +503,29 @@ func RegisterRoutes(m *macaron.Macaron) {
 				return
 			}
 
+			repository, unitType, err := attach.LinkedRepository()
+			if err != nil {
+				ctx.ServerError("LinkedRepository", err)
+				return
+			}
+
+			if repository == nil {
```

Comment:
```suggestion
			if repository == nil && (attach.IssueID != 0 || attach.ReleaseID != 0 || attach.CommentID != 0 || attach.UploaderID != ctx.User.ID) {
```

Reply:
If understand, the goal is to make it available to the uploader before linking the attachement ? 

If it is the case I will add a check for isLogged to be sure that ctx.User is ready.

Reply:
yes that was my intention. Anonymous user should not be allowed to upload but checking isLogged would not hurt :)

Reply:
Anon uploading attachement, is the last part of the #7956 to be made in a separate PR. But the check would be for a unlogged visitor on a page where someone would have paste the attachement link. On the same subject, we would also need an admin func to purge attachement not linked and uploaded since an elapsed time.

Reply:
That should be changed in other PR to account that than

Reply:
Let's keep PR small and fix one issue at a time :)